### PR TITLE
ctpdev: lumi inputs as parameters

### PR DESCRIPTION
**Pull request by lietava**
No other changes

### CODE REVIEW

**Comment by lietava** [comment]
@shahor02 : please, comments.

File: DataFormats/Detectors/CTP/include/DataFormatsCTP/LumiInfo.h [hunk 1]
```diff
@@ -27,11 +28,14 @@ struct LumiInfo {
...
   uint64_t counts = 0;      // counts in the interval for the nominal lumi detector (FT0)
   uint64_t countsFV0 = 0;   // connts for FV0 (less reliable)
+  int inp1 = 3;             // TVX
+  int inp2 = 6;             // VBA
   float getLumi() const { return nHBFCounted > 0 ? float(counts / (nHBFCounted * o2::constants::lhc::LHCOrbitMUS * 1e-6)) : 0.f; }
   float getLumiFV0() const { return nHBFCountedFV0 > 0 ? float(countsFV0 / (nHBFCountedFV0 * o2::constants::lhc::LHCOrbitMUS * 1e-6)) : 0.f; }
   float getLumiError() const { return nHBFCounted > 0 ? float(std::sqrt(counts) / (nHBFCounted * o2::constants::lhc::LHCOrbitMUS * 1e-6)) : 0.f; }
   float getLumiFV0Error() const { return nHBFCountedFV0 > 0 ? float(std::sqrt(countsFV0) / (nHBFCountedFV0 * o2::constants::lhc::LHCOrbitMUS * 1e-6)) : 0.f; }
-  ClassDefNV(LumiInfo, 2);
+  void printInputs() const { LOG(info) << "Lumi inp1:" << inp1 << ":" << o2::ctp::CTPInputsConfiguration::getInputNameFromIndex(inp1) << " inp2:" << inp2 << ":" << o2::ctp::CTPInputsConfiguration::getInputNameFromIndex(inp2); }
```
Comment (shahor02) [inline DataFormats/Detectors/CTP/include/DataFormatsCTP/LumiInfo.h]: Please move printInputs implementation and `#include "DataFormatsCTP/Configuration.h"` to cxx.

File: Detectors/CTP/reconstruction/src/RawDataDecoder.cxx [hunk 2]
```diff
@@ -310,6 +311,23 @@ int RawDataDecoder::decodeRaw(o2::framework::InputRecord& inputs, std::vector<o2
   return ret;
 }
 //
+int RawDataDecoder::setLumiInp(int lumiinp, std::string inp)
+{
+  // check if valis input
+  int index = o2::ctp::CTPInputsConfiguration::getInputIndexFromName(inp);
+  if (index == 0xff) {
+    return 0xff;
```
Comment (shahor02) [inline Detectors/CTP/reconstruction/src/RawDataDecoder.cxx:322]: I would produce fatal if wrong input name is passed.

**Review by shahor02 — COMMENTED** [review]
Looks fine, just 2 minor comments below.

File: DataFormats/Detectors/CTP/include/DataFormatsCTP/LumiInfo.h (hunk 1 above)
Comment (lietava) [reply DataFormats/Detectors/CTP/include/DataFormatsCTP/LumiInfo.h]: Hi @shahor02 , I was trying to do this. But I got error  /home/rl/alice/sw/SOURCES/O2/ctpdev/0/DataFormats/Detectors/CTP/src/LumiInfo.cxx:17:6: error: 'LumiInfo' has not been declared
   17 | void LumiInfo::printInputs() const
Either I overlook something trivial or I dont know. I spent some time with it, I am giving up, please, have a look.

Comment (lietava) [reply DataFormats/Detectors/CTP/include/DataFormatsCTP/LumiInfo.h]: I mean I pushed I expect it would not compile with above error.

Comment (shahor02) [reply DataFormats/Detectors/CTP/include/DataFormatsCTP/LumiInfo.h]: Well, you should leave the declaration `void printInputs() const;` in the `LumiInfo.h`, just the implementation with needed includes goes to cxx.

Comment (lietava) [reply DataFormats/Detectors/CTP/include/DataFormatsCTP/LumiInfo.h]: @shahor02 : declaration is left in LumiInfo.h. It looks like LumiInfo.h is not included in LumiInfo.cxx although there is include. Some problem with ifndef ...?

Comment (lietava) [reply DataFormats/Detectors/CTP/include/DataFormatsCTP/LumiInfo.h]: and actually I forgot to push latest version where I tried to clean includes without succes. Pushing now,

File: DataFormats/Detectors/CTP/src/LumiInfo.cxx [hunk 3]
```diff
@@ -0,0 +1,20 @@
...
+//
+// In applying this license CERN does not waive the privileges and immunities
+// granted to it by virtue of its status as an Intergovernmental Organization
+// or submit itself to any jurisdiction.
+
+/// \file LumiInfo.cxx
+
+#include "DataFormatsCTP/LumiInfo.h"
+// #include "DataFormatsCTP/Configuration.h"
+
```
Comment (shahor02) [inline DataFormats/Detectors/CTP/src/LumiInfo.cxx:19]: You forgot the namespace, add `using namespace o2::ctp;`

File: Detectors/CTP/reconstruction/src/RawDataDecoder.cxx (hunk 2 above)
Comment (lietava) [reply Detectors/CTP/reconstruction/src/RawDataDecoder.cxx:322]: done

File: DataFormats/Detectors/CTP/src/LumiInfo.cxx (hunk 3 above)
Comment (lietava) [reply DataFormats/Detectors/CTP/src/LumiInfo.cxx:19]: yes, thanks.

File: DataFormats/Detectors/CTP/include/DataFormatsCTP/LumiInfo.h (hunk 1 above)
Comment (lietava) [reply DataFormats/Detectors/CTP/include/DataFormatsCTP/LumiInfo.h]: done

**Comment by lietava** [comment]
@shahor02 , please, merge.

**Pull request merged**